AEP/aep-00004.md

```diff
@@ -0,0 +1,187 @@
+Using a database instead of (or in addition to) a cache would avoid too many discoveries on startups, but could limit the scaling to what the database can handle.
+
+It would be possible to enable communications between Load-Balancers in order to reduce the number of discoveries altogether, at the price of a much more complicated Load Balancer.
+
+### Authentication
+
+There are many ways to handle the authentication in the presence of load balancing.
+Whatever solution is chosen, the TLS connection must be terminated at the Load Balancer to let it interpret the queries.
+
+One possibility would be to perform authentication and authorization at the Load Balancer level and forward the request to the cluster only if the user is permitted to do the action.
+Authentication would be disabled on the clusters, or at least would allow the Load Balancer to perform any action.
+This would greatly complexify the Load Balancer itself.
+
+Another possibility would be to let the Load Balancer impersonate users to the cluster using the standard impersonate feature of the authentication system.
+This would force the Load Balancer to have a dedicated user on the clusters with the permission to impersonate any user.
+As it is currently not possible to chain impersonates, it would also forbid end-users to impersonate themselves, even if they have the permission to do it.
+
+Last possibility would be to make the Load Balancer forward the identity headers to the clusters, in the same way the nginx ingress transmits the identity to the control plane currently.
+Extra configuration at the cluster ingress would be required to allow the Load Balancer to set those sensitive headers.
+This last solution is the preferred one as it is simple to implement, secure, and featureful.
+
+### Partitions
+
+As a single session cannot span multiple clusters, all the partitions requested by the session must exist on the target cluster.
+The Load Balancer would be able to select a cluster that supports all the partitions requested, if any.
+It should avoid selecting a cluster that does not support a partition, and if no cluster supports all the partitions, the Load Balancer would raise an error.
+
+As an example, consider a cluster A with partitions p1 and p2, and a cluster B with partitions p2 and p3.
+If a session needs to target p2, it can go on either A or B.
+If a session targets p1 and p2, it can only go on A.
+If a session targets p3, it can only go on B.
+If a session targets p1 and p3, it cannot go anywhere, and the Load Balancer raises an error.
+If a session targets p4, the Load Balancer raises an error.
+
+### Cluster health
+
+With a static mapping between sessions and clusters (eg: without a database), if the chosen cluster is not reachable at the time the cluster is chosen, an error is raised.
+There is no internal retry for static mappings.
```

Comment:
What happens if multiple clusters could host that session ? If cluster A or B can host the session, would we try on A, and if it fails, try on B ?